### PR TITLE
EventManager::removeEventListener: fixed emptying list of sorted events

### LINKED ISSUE #80: Issue when manually removing subscriber and have multiple subscribers on the same event
I have some functional tests and need to disable a few subscribers that are not important for the test environment. The only way that I figured out was to manually remove them during the test setup. 

It looks something like this

``` php
$container = $this->getContainer();
$evm = $this->getEventManager();

$subscribers = [
    BlameableSubscriber::class,
    NotificationSubscriber::class,
];

foreach ($subscribers as $class) {
    /** @var Subscriber $subscriber */
    $subscriber = $container->getByType($class);
    $evm->removeEventListener($subscriber->getSubscribedEvents(), $subscriber);
}
```

I found a bug tough.

* Let's say I have event `onSave`
* I also have 2 subscribers, that subscribe to this event - `NotificationSubscriber` and `AnotherSubscriber`

I remove `NotificationSubscriber` from the manager as shown above, which means, that only `AnotherSubscriber` remains.

Problem is, when removing `NotificationSubscriber`, there's this code in `Kdyby\Events\EventManager` on line 222

``` php
foreach ((array) $unsubscribe as $eventName) {
	foreach ($this->listeners[$eventName] as $priority => $listeners) {
		if (($key = array_search($subscriber, $listeners, TRUE)) === FALSE) {
			continue;
		}

		unset($this->listeners[$eventName][$priority][$key]);
		if (empty($this->listeners[$eventName][$priority])) {
			unset($this->listeners[$eventName][$priority]);
		}
		if (empty($this->listeners[$eventName])) {
			unset($this->listeners[$eventName]);
		}

		// there are no listeners for this specific event, so no reason to call sort on next dispatch
		$this->sorted[$eventName] = array();
	}
}
```

Particularly, the issue is with this line

``` php
// there are no listeners for this specific event, so no reason to call sort on next dispatch
$this->sorted[$eventName] = array();
```

This effectively causes that `$this->sorted['onSave']` is set to an empty array, even tough it should still contain `AnotherSubscriber`. Later, when the event is dispatched, the listeners are fetched from this array, which results in no subscribers being called, event tough `AnotherSubscriber` is still registered in `Kdyby\Events\EventManager::listeners`.

### PR DESCRIPTION
**Pull request by matej21**
closes #80, #81

### CODE REVIEW

File: src/Kdyby/Events/EventManager.php [hunk 1]
```diff
@@ -222,10 +222,13 @@ public function removeEventListener($unsubscribe, $subscriber = NULL)
 				}
 				if (empty($this->listeners[$eventName])) {
 					unset($this->listeners[$eventName]);
+					// there are no listeners for this specific event, so no reason to call sort on next dispatch
+					$this->sorted[$eventName] = array();
+				} else {
+					// otherwise it needs to be sorted again
+					unset($this->sorted[$eventName]);
```
Comment (fprochazka) [inline src/Kdyby/Events/EventManager.php]: does it? If it's empty then it should return an empty array, no?

Comment (fprochazka) [inline src/Kdyby/Events/EventManager.php]: I see now :)

**Comment by fprochazka** [comment]
Thank you!

**Comment by matej21** [comment]
:+1:

**Pull request merged**